cddl/examples/comid-cend-inst-copy.diag

```diff
@@ -0,0 +1,84 @@
+/ concise-mid-tag / {	
+  / tag-identity / 1 : {	
+    / tag-id / 0 : "my-ns:acme-roadrunner-supplement"	
+  },	
+  / entity / 2 : [ {	
+    / entity-name / 0 : "ACME Inc.",	
+    / reg-id / 1 : 32("https://acme.example"),	
+    / role / 2 : [ 1,0,2 ] / creator, tag-creator, maintainer /	
+  } ],	
+  / triples / 4 : {	
+    / conditional-endorsement-triples / 10 : [	
+    [
+      [
+        [ / *** stateful-environment-record entry #0 *** /	
+          / environment-map / {	
+            / class / 0 : {
+              / class-id / 0 : / tagged-oid-type / 111(h'5502C000'),	
+              / vendor / 1 : "ACME Inc.",	
+              / model / 2 : "ACME RoadRunner Bootloader",
+              / layer / 3 : 0
+            }
+            / *** instance not present, so matches against any instance *** /
+          }, 
+          [         	
+            { / *** measurement-map *** /	
+              / mval / 1 : / measurement-values-map / {	
+                / comid.digests / 2 : [ [
+                  / hash-alg-id / 1, / sha256 /
+                  / hash-value / h'879432e53dd6571c7fa9bccafb75f488259262d6ea3a4d91b44aa336af4cb14a'
+                ] ]
+              }	
+            }	
+          ]
+        ], 
+        [ / *** stateful-environment-record entry #1 *** /
+          / environment-map / {
+            / class / 0 : {
+              / class-id / 0 : / tagged-oid-type / 111(h'5502C000'),
+              / vendor / 1 : "ACME Inc.",
+              / model / 2 : "ACME RoadRunner Firmware",
+              / layer / 3 : 1
+            },
+            / instance / 1 : 565(0) / *** Match against ACS entry with instance which stateful-environment-record #0 matched *** /
```

Comment:
Does instance zero (i.e. 565(0)) have special significance? Otherwise, how does Endorser know the instance id if it was dynamically asserted into the Evidence / ACS?

Reply:
Yes, as a condition, this is a binding position for a variable named 0.
As an endorsement, 565(0) is meant to be replaced with the instance that was bound to that variable during condition matching.